chew.go

```diff
@@ -41,17 +77,126 @@ var contentTypeProcessors = map[string]func(io.Reader, string) ([]common.Chunk,
 	contentTypeEPUB:     document.ProcessEpub,
 }
 
+var (
+	rateLimiter       *rate.Limiter
+	robotsCache       = make(map[string]*robotstxt.RobotsData)
+	robotsMu          sync.RWMutex
+	userAgent         = "Chew/1.0 (+https://github.com/mmatongo/chew)"
+	lastAccessTimes   = make(map[string]time.Time)
+	lastAccessMu      sync.Mutex
+	retryLimit        = 3
+	retryDelay        = 5 * time.Second
+	crawlDelay        = 10 * time.Second
+	proxyList         []string
+	currentProxyIndex int
+	proxyMu           sync.Mutex
+	ignoreRobotsTxt   bool
+)
+
+/*
+NewConfig allows you to set the configuration options for URL processing. It takes a Config struct.
+
+Usage:
+
+	config := chew.Config{
+		UserAgent:       "MyBot/1.0 (+https://example.com/bot)",
+		RetryLimit:      3,
+		RetryDelay:      5 * time.Second,
+		CrawlDelay:      10 * time.Second,
+		ProxyList:       []string{"http://proxy1.com", "http://proxy2.com"},
+		RateLimit:       rate.Every(2 * time.Second),
+		RateBurst:       3,
+		IgnoreRobotsTxt: false,
+	}
+
+	chew.NewConfig(config)
+*/
+func NewConfig(config common.Config) {
+	userAgent = config.UserAgent
+	retryLimit = config.RetryLimit
+	retryDelay = config.RetryDelay
+	crawlDelay = config.CrawlDelay
+	proxyList = config.ProxyList
+	rateLimiter = rate.NewLimiter(config.RateLimit, config.RateBurst)
+	ignoreRobotsTxt = config.IgnoreRobotsTxt
+}
+
+func init() {
+	NewConfig(common.Config{
+		UserAgent:       userAgent,
+		RetryLimit:      3,
+		RetryDelay:      5 * time.Second,
+		CrawlDelay:      10 * time.Second,
+		ProxyList:       []string{},
+		RateLimit:       rate.Every(2 * time.Second),
+		RateBurst:       3,
+		IgnoreRobotsTxt: false,
+	})
+}
+
 /*
-Transcribe uses the Google Cloud Speech-to-Text API to transcribe an audio file. It takes
-a context, the filename of the audio file to transcribe, and a TranscribeOptions struct which
-contains the Google Cloud credentials, the GCS bucket to upload the audio file to, and the language code
-to use for transcription. It returns the transcript of the audio file as a string and an error if the
-transcription fails.
+Transcribe is a function that transcribes audio files using either the Google Cloud Speech-to-Text API
+or the Whisper API. It handles uploading the audio file to Google Cloud Storage if necessary,
+manages the transcription process, and returns the resulting transcript.
+
+For detailed usage instructions, see the TranscribeOptions struct documentation.
 */
 var Transcribe = transcribe.Transcribe
 
+/*
+The TranscribeOptions struct contains the options for transcribing an audio file. It allows the user
+to specify the Google Cloud credentials, the GCS bucket to upload the audio file to, the language code
+to use for transcription, a potion to enable diarization including the min and max speakers and
```

Comment:
maybe something to explain what is diarization

```suggestion
to use for transcription, a potion to enable diarization (blah blah...) including the min and max speakers and
```